src/main/php/PHPMD/ProcessingError.php

```diff
@@ -73,19 +73,13 @@ public function getMessage()
      * Evil hack that extracts the source file from the original exception
      * message. This method should be removed once we have added the source file
      * as a mandatory property to PDepend's exceptions.
-     *
-     * @param string $message
-     * @return string
      */
-    private function extractFile($message)
+    private function extractFile(?string $message): string
```

Comment:
As best I can tell this should never receive a null. It's being fed from the constructor and the object is only ever created from an exception getMessage() which always returns a string.
```suggestion
    private function extractFile(string $message): string
```